calendar-bundle/src/EventListener/DataContainer/EventSearchListener.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\EventListener\DataContainer;
+
+use Contao\CalendarEventsModel;
+use Contao\CalendarModel;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Routing\ContentUrlGenerator;
+use Contao\DataContainer;
+use Contao\PageModel;
+use Contao\Search;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * @internal
+ */
+class EventSearchListener
+{
+    public function __construct(
+        private readonly ContaoFramework $framework,
+        private readonly ContentUrlGenerator $urlGenerator,
+    ) {
+    }
+
+    #[AsCallback(table: 'tl_calendar_events', target: 'fields.alias.save')]
+    public function onSaveAlias(string $value, DataContainer $dc): string
+    {
+        if ($value === ($dc->getCurrentRecord()['alias'] ?? null)) {
+            return $value;
+        }
+
+        $this->purgeSearchIndex((int) $dc->id);
+
+        return $value;
+    }
+
+    #[AsCallback(table: 'tl_calendar_events', target: 'fields.robots.save')]
+    public function onSaveRobots(string $value, DataContainer $dc): string
+    {
+        if ($dc->getCurrentRecord()['robots'] === $value || str_starts_with($value, 'index')) {
+            return $value;
+        }
+
+        if ('' === $value && str_starts_with($dc->getCurrentRecord()['robots'], 'index')) {
+            // Get robots tag of the reader page (linked in calendar)
+            $readerPageId = $this->framework->getAdapter(CalendarModel::class)->findById($dc->getCurrentRecord()['pid'])->jumpTo ?? null;
+
+            if ($readerPageId) {
+                $readerPageRobots = $this->framework->getAdapter(PageModel::class)->findById($readerPageId)->robots ?? '';
```

Comment:
If you inject `Doctrine\DBAL\Connection` this could be shortened to:

```php
$readerPageRobots = $this->connection->fetchOne('SELECT p.robots FROM tl_page AS p, tl_calendar AS c WHERE c.id = ? AND c.jumpTo = p.id', [$dc->getCurrentRecord()['pid']]);
```